web/source/dom/targets/touchAlias.ts

```diff
@@ -25,15 +31,29 @@ namespace com.keyman.dom.targets {
     }
 
     hasSelection(): boolean {
-      // Always has an internal caret position.
-      return true;
+      let Lsel = this.root.ownerDocument.getSelection();
+      let outerSel = document.getSelection();
+
+      if (Lsel && outerSel && Lsel.toString().length > 0) {
+        // If the outer doc's selection matches, we're active.
+        // Both start and end elements need to be the same
+        if(outerSel.anchorNode == Lsel.anchorNode && outerSel.focusNode == Lsel.focusNode &&
+            Lsel.anchorNode == Lsel.focusNode) {
+          return true;
+        }
+      }
+      return false; // designIFrame returns true?
     }
 
     getDeadkeyCaret(): number {
       return this.root.getTextCaret();
     }
 
     getTextBeforeCaret(): string {
+      if (this.hasSelection()) {
+        // If there's selected text, use nul for context
+        return '';
+      }
```

Comment:
This change needs to be elsewhere I think -- there are multiple `getTextBeforeCaret()` functions and they all need to be handling this situation.

Note that contentEditable.ts has the exact opposite of this test.

I think `hasSelection()` is returning true if the caret is in the text area even if there is no actual selection. But I would have to dig deeper to be sure.